cow_instrument/InstrumentTree.cpp

```diff
@@ -18,53 +18,68 @@ void findDetectors(const Component &component,
 }
 
 /*
- * Derived, detector vector is always recreated.
+ * Derived, detector vector is always recreated. Except on a plain-old copy.
 */
-InstrumentTree::InstrumentTree(std::vector<Node> &&nodes, size_t nDetectors)
-    : m_nodes(std::move(nodes)) {
-
+void InstrumentTree::init(size_t nDetectors) {
   // This will make the push_backs faster
-  m_detectorVec.reserve(nDetectors);
+  m_detectorVec->reserve(nDetectors);
```

Comment:
When we modify the instrument, the capacity is correct already (after `m_detectorVec->clear();` below).

So the only case where this is relevant is creating a new instrument. This is more rare, and there are so many other things going on at the same time that the automatic growth mechanism should be fast enough (and I would be surprised if we saw a big difference).